solarforecastarbiter/metrics/probabilistic.py

```diff
@@ -528,19 +538,37 @@ def continuous_ranked_probability_score(obs, fx, fx_prob):
     elif np.shape(fx)[1] < 2:
         raise ValueError("forecasts must have d >= 2 CDF intervals "
                          f"(expected >= 2, got {np.shape(fx)[1]})")
-    else:
-        obs = np.tile(obs, (fx.shape[1], 1)).T
 
-    # event: 0=did not happen, 1=did happen
-    o = np.where(obs <= fx, 1.0, 0.0)
+    n = len(fx)
+
+    # extend CDF min to ensure obs within forecast support
+    # (n, d) ==> (n, d + 1)
+    fx_min = np.minimum(obs, fx[:, 0])
+    fx = np.hstack([fx_min[:, np.newaxis], fx])
+    fx_prob = np.hstack([np.zeros([n, 1]), fx_prob])
```

Comment:
`np.zeros` because we're assuming the left-hand side of the CDF goes to zero, i.e., that whether the `fx_min` is `obs` or `fx[:, 0]` the corresponding probability is zero. Also, if `fx_min = fx[:, 0]` but `fx_prob[:, 0] != 0`, this operation means the integration is done with a zero-width area and therefore doesn't change the CRPS result (`fx[:, 1] - fx[:, 0] = 0` since now `fx[:, 1] = fx[:, 0]`).